cookbook/providers/agent.rb

```diff
@@ -0,0 +1,111 @@
+action :create do
+  node.set[:baragon][:agent_yaml]['loadBalancerConfig']['name'] = new_resource.group
+
+  agent_root_path = "#{node['baragon']['agent_yaml']['loadBalancerConfig']['rootPath']}/#{new_resource.group}"
+
+  node.set[:baragon][:agent_yaml]['server']['connector']['port'] = new_resource.port
+
+  # rubocop:disable Metrics/LineLength
+  ["#{agent_root_path}/proxy",
+   "#{agent_root_path}/upstreams"].each do |dir|
+    directory dir do
+      recursive true
+    end
+  end
+  # rubocop:enable Metrics/LineLength
+
+  case node[:baragon][:install_type]
+  when 'source'
+    run_context.include_recipe 'baragon::build'
+
+    remote_file "/usr/share/java/BaragonAgentService-#{node[:baragon][:version]}-shaded.jar" do
+      action :create
+      backup 5
+      owner 'root'
+      group 'root'
+      mode 0644
+      source "file://#{Chef::Config[:file_cache_path]}/Baragon/" \
+             'BaragonAgentService/target/' \
+             "BaragonAgentService-#{node[:baragon][:version]}-SNAPSHOT-shaded.jar"
+    end
+  when 'package'
+    run_context.include_recipe 'maven'
+
+    maven 'BaragonAgentService' do
+      group_id 'com.hubspot'
+      classifier 'shaded'
+      version node['baragon']['version']
+      dest '/usr/share/java'
+    end
+  else
+    fail "Unsupported install type: #{node[:baragon][:install_type]}"
+  end
+
+  node.set[:baragon][:agent_yaml][:zookeeper][:quorum] =
+    node[:baragon][:zk_hosts].join(',')
+  node.set[:baragon][:agent_yaml][:zookeeper][:zkNamespace] =
+    node[:baragon][:zk_namespace]
+
+  if node[:nginx]
```

Comment:
I think we might want to have the `nginx` related section separated out of this for a few reasons:
- If someone is installing this, it isn't unreasonable that they will either set `checkConfigCommand` and `reloadConfigCommand` or have a different way of installing nginx where `[:nginx][:binary]` might not be set. `/bin/true` can always be the default for those
- While Baragon was initially built around nginx, we don't want to restrict it to nginx (There are ways to make it work for haproxy too).
@tpetr feel free to jump in here if you disagree ^